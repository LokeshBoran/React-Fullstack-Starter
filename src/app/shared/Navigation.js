import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { routerActions } from 'react-router-redux';
import { AppBar, Toolbar, IconButton, Icon, Typography } from 'material-ui';

import Link from '~/shared/Link';
import Button from '~/shared/Button';

export const _Navigation = () => {

  return (
    <div>
      <AppBar>
        <Toolbar>
          {
            location.pathname === '/'
              ? <IconButton color="contrast" aria-label="Menu">
                  <Icon>toys</Icon>
                </IconButton>
              : <Link to="/">
                  <IconButton color="contrast">
                    <Icon>arrow_back</Icon>
                  </IconButton>
                </Link>
          }
          <Typography type="title" color="inherit">
            React by Example, location: <span className="location">{location.pathname}</span>.
          </Typography>
        </Toolbar>
      </AppBar>

      <div className="filling"></div>

      {
        location.pathname === '/'
          ? <div className="buttons">
              <Link to="/counter">
                <Button raised color="teal">Counter</Button>
              </Link>
              {'　'}
              <Link to="/crud">
                <Button raised color="teal">CRUD</Button>
              </Link>
              {'　'}
              <Link to="/rest">
                <Button raised color="teal">REST</Button>
              </Link>
              {'　'}
              <Link to="/graphql">
                <Button raised color="teal">GraphQL</Button>
              </Link>
              {'　'}
              <Link to="/form-controls">
                <Button raised color="teal">Form Controls</Button>
              </Link>
              {'　'}
              <Link to="/data-table">
                <Button raised color="teal">Data Table</Button>
              </Link>
              {'　'}
              <Link to="/authorization">
                <Button raised color="teal">Authorization</Button>
              </Link>
            </div>
          : void 0
      }

      <style jsx>{`
        .location {
          color: #F48FB1;
        }

        .filling {
          padding-top: 64px;
        }

        .buttons {
          display: inline-flex;
        }
      `}</style>
    </div>
  );
};

export default connect(
  ({ router }) => ({ router }),
  dispatch => ({ actions: bindActionCreators(routerActions, dispatch) })
)(_Navigation);
