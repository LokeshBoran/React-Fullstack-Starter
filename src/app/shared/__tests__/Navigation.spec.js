import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { _Navigation } from '../Navigation';

configure({ adapter: new Adapter() });

describe('Navigation', () => {
  it('should render initial component', () => {
    const component = shallow(
      <_Navigation />
    );

    expect(component).toMatchSnapshot();
  });
});
