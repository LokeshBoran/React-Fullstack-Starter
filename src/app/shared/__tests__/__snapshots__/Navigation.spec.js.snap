// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render initial component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(AppBar)>
          <WithStyles(Toolbar)>
            <Unknown
              to="/"
            >
              <WithStyles(IconButton)
                color="contrast"
              >
                <WithStyles(Icon)>
                  arrow_back
                </WithStyles(Icon)>
              </WithStyles(IconButton)>
            </Unknown>
            <WithStyles(Typography)
              color="inherit"
              type="title"
            >
              React by Example, location: 
              <span
                className="jsx-3411527522 location"
              >
                blank
              </span>
              .
            </WithStyles(Typography)>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>,
        <div
          className="jsx-3411527522 filling"
        />,
        false,
        <JSXStyle
          css=".location.jsx-3411527522{color:#f48fb1;}.filling.jsx-3411527522{padding-top:64px;}.buttons.jsx-3411527522{display:-webkit-inline-box;display:-ms-inline-flexbox;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}"
          styleId="3411527522"
        />,
      ],
      "className": "jsx-3411527522",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <Unknown
              to="/"
            >
              <WithStyles(IconButton)
                color="contrast"
              >
                <WithStyles(Icon)>
                  arrow_back
                </WithStyles(Icon)>
              </WithStyles(IconButton)>
            </Unknown>
            <WithStyles(Typography)
              color="inherit"
              type="title"
            >
              React by Example, location: 
              <span
                className="jsx-3411527522 location"
              >
                blank
              </span>
              .
            </WithStyles(Typography)>
          </WithStyles(Toolbar)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown
                to="/"
              >
                <WithStyles(IconButton)
                  color="contrast"
                >
                  <WithStyles(Icon)>
                    arrow_back
                  </WithStyles(Icon)>
                </WithStyles(IconButton)>
              </Unknown>,
              <WithStyles(Typography)
                color="inherit"
                type="title"
              >
                React by Example, location: 
                <span
                  className="jsx-3411527522 location"
                >
                  blank
                </span>
                .
              </WithStyles(Typography)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <WithStyles(IconButton)
                  color="contrast"
                >
                  <WithStyles(Icon)>
                    arrow_back
                  </WithStyles(Icon)>
                </WithStyles(IconButton)>,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Icon)>
                    arrow_back
                  </WithStyles(Icon)>,
                  "color": "contrast",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "arrow_back",
                  },
                  "ref": null,
                  "rendered": "arrow_back",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "React by Example, location: ",
                  <span
                    className="jsx-3411527522 location"
                  >
                    blank
                  </span>,
                  ".",
                ],
                "color": "inherit",
                "type": "title",
              },
              "ref": null,
              "rendered": Array [
                "React by Example, location: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "blank",
                    "className": "jsx-3411527522 location",
                  },
                  "ref": null,
                  "rendered": "blank",
                  "type": "span",
                },
                ".",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "jsx-3411527522 filling",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".location.jsx-3411527522{color:#f48fb1;}.filling.jsx-3411527522{padding-top:64px;}.buttons.jsx-3411527522{display:-webkit-inline-box;display:-ms-inline-flexbox;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}",
          "styleId": "3411527522",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(AppBar)>
            <WithStyles(Toolbar)>
              <Unknown
                to="/"
              >
                <WithStyles(IconButton)
                  color="contrast"
                >
                  <WithStyles(Icon)>
                    arrow_back
                  </WithStyles(Icon)>
                </WithStyles(IconButton)>
              </Unknown>
              <WithStyles(Typography)
                color="inherit"
                type="title"
              >
                React by Example, location: 
                <span
                  className="jsx-3411527522 location"
                >
                  blank
                </span>
                .
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>,
          <div
            className="jsx-3411527522 filling"
          />,
          false,
          <JSXStyle
            css=".location.jsx-3411527522{color:#f48fb1;}.filling.jsx-3411527522{padding-top:64px;}.buttons.jsx-3411527522{display:-webkit-inline-box;display:-ms-inline-flexbox;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}"
            styleId="3411527522"
          />,
        ],
        "className": "jsx-3411527522",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <Unknown
                to="/"
              >
                <WithStyles(IconButton)
                  color="contrast"
                >
                  <WithStyles(Icon)>
                    arrow_back
                  </WithStyles(Icon)>
                </WithStyles(IconButton)>
              </Unknown>
              <WithStyles(Typography)
                color="inherit"
                type="title"
              >
                React by Example, location: 
                <span
                  className="jsx-3411527522 location"
                >
                  blank
                </span>
                .
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Unknown
                  to="/"
                >
                  <WithStyles(IconButton)
                    color="contrast"
                  >
                    <WithStyles(Icon)>
                      arrow_back
                    </WithStyles(Icon)>
                  </WithStyles(IconButton)>
                </Unknown>,
                <WithStyles(Typography)
                  color="inherit"
                  type="title"
                >
                  React by Example, location: 
                  <span
                    className="jsx-3411527522 location"
                  >
                    blank
                  </span>
                  .
                </WithStyles(Typography)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <WithStyles(IconButton)
                    color="contrast"
                  >
                    <WithStyles(Icon)>
                      arrow_back
                    </WithStyles(Icon)>
                  </WithStyles(IconButton)>,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(Icon)>
                      arrow_back
                    </WithStyles(Icon)>,
                    "color": "contrast",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "arrow_back",
                    },
                    "ref": null,
                    "rendered": "arrow_back",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "React by Example, location: ",
                    <span
                      className="jsx-3411527522 location"
                    >
                      blank
                    </span>,
                    ".",
                  ],
                  "color": "inherit",
                  "type": "title",
                },
                "ref": null,
                "rendered": Array [
                  "React by Example, location: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "blank",
                      "className": "jsx-3411527522 location",
                    },
                    "ref": null,
                    "rendered": "blank",
                    "type": "span",
                  },
                  ".",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "jsx-3411527522 filling",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".location.jsx-3411527522{color:#f48fb1;}.filling.jsx-3411527522{padding-top:64px;}.buttons.jsx-3411527522{display:-webkit-inline-box;display:-ms-inline-flexbox;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}",
            "styleId": "3411527522",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
